src/surface/transceiver/transceiver/serial_reader.py

```diff
@@ -1,45 +1,93 @@
+                                                    QoSPresetProfiles.SENSOR_DATA.value)
+
+        self.create_subscription(FloatCommand, 'float_command', self.send_command,
+                                 QoSPresetProfiles.DEFAULT.value)
+
+        self.serial_publisher = self.create_publisher(FloatSerial, 'float_serial',
+                                                      QoSPresetProfiles.SENSOR_DATA.value)
         timer_period = .5
-        self.ser = serial.Serial('/dev/ttyTransceiver', 115200)
-        self.timer = self.create_timer(timer_period, self.timer_callback)
-        self.i = 0
+
+        self.first_attempt = True
+        self.create_timer(timer_period, self.timer_callback)
+        try:
+            with Serial("/dev/serial/by-id/usb-Adafruit_Feather_32u4-if00", 115200) as ser:
+                self.serial = ser
+                self.get_logger().info("Serial device connected.")
+        except SerialException as e:
+            self.get_logger().error("Error no transceiver connected.")
+            self.get_logger().error(str(e))
+            exit(1)
+
+    def send_command(self, msg: FloatCommand) -> None:
+        try:
+            with self.serial:
+                self.serial.write(msg.command.encode())
+        except SerialException as e:
+            self.get_logger().error("Command send failed.")
+            self.get_logger().error(str(e))
 
     def timer_callback(self) -> None:
         """Publish a message from the transceiver."""
-        msg = FloatCommand()
-        msg.command = self.ser.readline().decode()
-        self.publisher.publish(msg)
-
-    def control_callback(self, msg: FloatCommand) -> None:
-        """
-        Log a binary string of the command that was sent.
-
-        Parameters
-        ----------
-        msg : FloatCommand
-            the command that was sent via serial monitor
-
-        """
-        msg_encode: bytes = msg.command.encode()
-        self.ser.write(msg_encode)
-        # !r is to print a binary string
-        self.get_logger().info(f'Command sent via serial monitor: {msg_encode!r}')
+        try:
+            with self.serial:
+                packet = self.serial.readline().decode().strip()
+        except SerialException as e:
+            self.get_logger().error("Serial read failed.")
+            self.get_logger().error(str(e))
+            return
+
+        self.serial_publisher.publish(FloatSerial(serial=packet))
+
+        if ROS_PACKET not in packet:
+            return
+
+        msg = FloatData()
+
+        packet_sections = packet.split(SECTION_SEPARATOR)
+        header = packet_sections[1]
+        data = packet_sections[2]
+
+        msg.team_number = int(header[0])
+        msg.profile_number = int(header[1])
+        msg.profile_half = int(header[2])
+
+        time_data_list: list[float] = []
+        depth_data_list: list[float] = []
+        for time_data, depth_data in [data.split(COMMA_SEPARATOR) for data in
```

Comment:
Too many things called similar things
```suggestion
        for time_reading, depth_reading in [data.split(COMMA_SEPARATOR) for data in
```